cypress/integration/unit_tests/polygon_draw_test.js

```diff
@@ -49,142 +67,182 @@ describe('Unit test for ShapeData', () => {
 
-    // Set up appropriate Firestore mocks.
-    cy.stub(getFirestoreRoot(), 'collection')
-        .withArgs('usershapes')
-        .returns(firebaseCollection);
-    for (const prop in firebaseCollection) {
-      if (firebaseCollection.hasOwnProperty(prop)) {
-        delete firebaseCollection[prop];
-      }
-    }
-    // Make sure .get() succeeds.
-    firebaseCollection.get = () => [];
-    // Make sure userShapes is set in the code, and use our custom EarthEngine
-    // FeatureCollections.
-    return cy.wrap(initializeAndProcessUserRegions(null, Promise.resolve({
-      data: () => ({asset_data: {damage_asset_path: damageCollection}}),
-    })));
+    return setUpPage();
   });
 
-  it('Add shape', () => {
-    const popup = new StubPopup();
-    const underTest = new StoredShapeData(null, null, null, popup);
-    const records = [];
-    firebaseCollection.add = recordRecord(records, {id: 'new_id'});
-    popup.notes = 'my notes';
-    cy.wrap(underTest.update()).then(() => {
-      expect(records).to.eql([{
-        calculatedData: calculatedData,
-        geometry: polygonGeometry,
-        notes: 'my notes',
-      }]);
-      expect(underTest.id).to.eql('new_id');
-      expect(StoredShapeData.pendingWriteCount).to.eql(0);
-    });
+  /**
+   * Initializes the Google Map; sets up the polygon drawing manager; and sets
+   * up user feature handling. No features are expected on first call, since the
+   * test Firestore database is empty.
+   * @return {Cypress.Chainable}
+   */
+  function setUpPage() {
+    return createGoogleMap()
+        .then((mapResult) => map = mapResult)
+        .then(() => setUpPolygonDrawing(map, Promise.resolve(null)))
+        .then((drawingManagerResult) => drawingManager = drawingManagerResult)
+        .then(() => {
+          userRegionData.clear();
+          return initializeAndProcessUserRegions(map, Promise.resolve({
+            data: () => ({asset_data: {damage_asset_path: damageCollection}}),
+          }));
+        });
+  }
+
+  it('Adds polygon', () => {
+    drawPolygonAndClickOnIt();
+    pressPopupButton('close');
   });
 
-  it('Update shape', () => {
-    const popup = new StubPopup();
-    popup.setCalculatedData(calculatedData);
-    const underTest =
-        new StoredShapeData('my_id', 'my notes', polygonGeometry, popup);
-    const records = [];
-    const ids = [];
-    firebaseCollection.doc = (id) => {
-      ids.push(id);
-      return {set: recordRecord(records, null)};
-    };
-    popup.notes = 'new notes';
-    cy.wrap(underTest.update()).then(() => {
-      expect(ids).to.eql(['my_id']);
-      expect(records).to.eql([{
-        calculatedData: calculatedData,
-        geometry: polygonGeometry,
-        notes: 'new notes',
-      }]);
-      expect(underTest.id).to.eql('my_id');
-      expect(StoredShapeData.pendingWriteCount).to.eql(0);
+  it('Updates notes', () => {
+    drawPolygonAndClickOnIt();
```

Comment:
maybe also add a polygon shape update here? make it a general update test

Reply:
Also fine with a separate test - just didn't think we were testing that anywhere!

Reply:
We were incidentally testing it in the test below that asserted that ee computation happened if geometry changed.